rnd/autogpt_builder/src/hooks/useCredentials.ts

```diff
@@ -0,0 +1,94 @@
+import { CustomNodeData } from "@/components/CustomNode";
+import AutoGPTServerAPI, {
+  BlockIOCredentialsSubSchema,
+  CredentialsMetaResponse,
+} from "@/lib/autogpt-server-api";
+import { useNodeId, useNodesData } from "@xyflow/react";
+import { useRouter } from "next/navigation";
+import { useCallback, useEffect, useMemo, useState } from "react";
+
+export default function useCredentials() {
+  const nodeId = useNodeId();
+
+  if (!nodeId) {
+    throw new Error("useCredentials must be within a CustomNode");
+  }
+
+  const router = useRouter();
+  const [savedApiKeys, setSavedApiKeys] = useState<CredentialsMetaResponse[]>([]);
+  const [savedOAuthCredentials, setSavedOAuthCredentials] = useState<
+    CredentialsMetaResponse[]
+  >([]);
+
+  const data = useNodesData(nodeId)!.data as CustomNodeData;
+  const credentials = data.inputSchema.properties.credentials as BlockIOCredentialsSubSchema
+
+  const api = useMemo(() => new AutoGPTServerAPI(process.env.NEXT_PUBLIC_AGPT_SERVER_URL!), []);
```

Comment:
This is already the default, so:
```suggestion
  const api = useMemo(() => new AutoGPTServerAPI(), []);
```